mirdata/datasets/queen.py

```diff
@@ -0,0 +1,245 @@
+        title (str): title of the track
+        track_id (str): track id
+
+    """
+    def __init__(self, track_id, data_home):
+        if track_id not in DATA.index['tracks']:
+            raise ValueError("{} is not a valid track ID in Queen".format(track_id))
+
+        self.track_id = track_id
+
+        self._data_home = data_home
+        self._track_paths = DATA.index['tracks'][track_id]
+        self.chords_path = utils.none_path_join(
+            [self._data_home, self._track_paths["chords"][0]]
+        )
+        self.keys_path = utils.none_path_join(
+            [self._data_home, self._track_paths["keys"][0]]
+        )
+        self.sections_path = os.path.join(
+            self._data_home, self._track_paths["sections"][0]
+        )
+        self.audio_path = os.path.join(self._data_home, self._track_paths["audio"][0])
+
+        self.title = os.path.basename(self._track_paths["sections"][0]).split(".")[0]
+
+    @utils.cached_property
+    def chords(self):
+        """ChordData: chord annotation"""
+        return load_chords(self.chords_path)
+
+    @utils.cached_property
+    def key(self):
+        """KeyData: key annotation"""
+        return load_key(self.keys_path)
+
+    @utils.cached_property
+    def sections(self):
+        """SectionData: section annotation"""
+        return load_sections(self.sections_path)
+
+    @property
+    def audio(self):
+        """(np.ndarray, float): audio signal, sample rate"""
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            section_data=[(self.sections, None)],
+            chord_data=[(self.chords, None)],
+            key_data=[(self.key, None)],
+            metadata={"artist": "The Queen", "title": self.title},
+        )
+
+
+def load_audio(audio_path):
+    """Load a Queen audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=None, mono=True)
+
+
+def load_chords(chords_path):
+    """Load Queen format chord data from a file
+
+    Args:
+        chords_path (str): path to chord annotation file
+
+    Returns:
+        (utils.ChordData): loaded chord data
+
+    """
+    if chords_path is None:
+        return None
+
+    if not os.path.exists(chords_path):
+        raise IOError("chords_path {} does not exist".format(chords_path))
+
+    start_times, end_times, chords = [], [], []
+    with open(chords_path, "r") as f:
+        dialect = csv.Sniffer().sniff(f.read(1024))
+        f.seek(0)
+        reader = csv.reader(f, dialect)
+        for line in reader:
```

Comment:
just a suggestion. this could be faster with numpy
from numpy import genfromtxt
my_data = genfromtxt('my_file.csv', delimiter=',')

Reply:
It is a copy/paste from beatles dataset because structure is the same